### ISSUE TITLE
`xcvradm`: want ability to hide modules from upstack software

### ISSUE THREAD

**Issue by Aaron-Hartwig** [oxidecomputer/transceiver-control#279]
We can find ourselves in the setting where we use the `xcvradm` binary from a Scrimlet's switch zone to debug. However, the very fact that a switch zone exists usually means `dendrite` is running and automatically handing control of transceivers over to the SDE. It would be helpful to be able to gracefully hide modules from `dendrite` using `xcvradm`.

**Comment by bnaecker** [comment]
We can currently do this from the switch zone using swadm, using the set-management-mode sub command. Is that sufficient?On 20 Sep 2024, at 07:19, Aaron Hartwig ***@***.***> wrote:﻿
We can find ourselves in the setting where we use the xcvradm binary from a Scrimlet's switch zone to debug. However, the very fact that a switch zone exists usually means dendrite is running and automatically handing control of transceivers over to the SDE. It would be helpful to be able to gracefully hide modules from dendrite using xcvradm.

**Comment by Aaron-Hartwig** [comment]
Oh! Yeah probably 😅

**Comment by Aaron-Hartwig** [comment]
@bnaecker I attempted `set-management-mode qsfpN manual` and was still seeing messages come in. It might be worth revalidating that is working?

**Comment by bnaecker** [comment]
Yeah, I can definitely take a look. All the command really does is report that the module is not present to the SDE, regardless of what the present pin says. So it's definitely possible for the module to continue to operate in some ways after hiding it. What messages did you see?
